controllers/extendeddaemonset_e2e_test.go

```diff
@@ -0,0 +1,311 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// +build e2e
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	datadoghqv1alpha1 "github.com/DataDog/extendeddaemonset/api/v1alpha1"
+	"github.com/DataDog/extendeddaemonset/controllers/testutils"
+	// +kubebuilder:scaffold:imports
+)
+
+// This test may take ~30s to run, check you go test timeout
+var _ = Describe("ExtendedDaemonSet Controller", func() {
+	const (
+		timeout     = 1 * time.Minute
+		longTimeout = 5 * time.Minute
+		interval    = 2 * time.Second
+	)
+
+	intString2 := intstr.FromInt(2)
+	intString10 := intstr.FromInt(10)
+
+	Context("Initial deployment", func() {
```

Comment:
I think we should have specific, dedicated tests if we are to introduce E2E with Cluster.

Reply:
Do you mean splitting this monolithic one into separate tests? There is still a need for the initial deployment/canary phase before you can do anything else. Concern is duration of tests as well if we repeat that on every split test.

Reply:
I meant the duplication of tests, I believe these tests are duplicated from the original file:

```
It("Should handle EDS"
It("Should do canary deployment"
It("Should add canary labels"
It("Should remove canary labels"
```

Reply:
Yes, so we can just wait for canary to be over. I will remove these.